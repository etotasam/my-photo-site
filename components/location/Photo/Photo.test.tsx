import React from "react";
import { render, screen } from "@testing-library/react";
import { Photo } from ".";

const image = {
  documentId: "string",
  width: 200,
  createAt: new Date(),
  height: 200,
  url: "https://picsum.photos/200",
  filename: "",
  id: "1",
};

const props = {
  location: image,
  index: 1,
  hasBreak: false,
};

describe(`Photo Component`, () => {
  it(`hasBreakの値での表示の切り替えの検証`, () => {
    const { queryByTestId, rerender, asFragment } = render(<Photo {...props} />);
    expect(queryByTestId(`breaked`)).toBeNull();
    expect(queryByTestId(`non-break`)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
    props.hasBreak = true;
    rerender(<Photo {...props} />);
    expect(queryByTestId(`breaked`)).toBeInTheDocument();
    expect(queryByTestId(`non-break`)).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});
