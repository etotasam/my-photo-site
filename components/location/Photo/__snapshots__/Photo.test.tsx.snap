// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo Component hasBreakの値での表示の切り替えの検証 1`] = `
<DocumentFragment>
  <li
    class="w-1/2 md:w-1/5 mb-5 inline-block"
    data-testid="non-break"
  />
</DocumentFragment>
`;

exports[`Photo Component hasBreakの値での表示の切り替えの検証 2`] = `
<DocumentFragment>
  <li
    class="w-1/2 md:w-1/5 mb-5 inline-block"
    data-testid="breaked"
  >
    <a
      class="block cursor-pointer relative w-[90%] pt-[45%] mx-auto"
      style="opacity: 0; transform: translateY(20px) translateZ(0);"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt=""
          class="pointer-events-none"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
        <noscript />
      </div>
    </a>
    <h2
      class="font-extralight"
      data-testid="h2"
    >
      1
    </h2>
  </li>
</DocumentFragment>
`;
